doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,898 @@
+
+    Returns:
+        An integer, the index of the ModuleBase object in the module_list
+```
+#### 3.1.2 ChassisBase class new APIs
+is_smartswitch(self):
+```
+    Retrieves whether the sonic instance is part of smartswitch
+
+    Returns:
+      True
+```
+
+get_module_dpu_port(self, index):
+```
+    Retrieves the DPU port - internal ASIC port for DPU represented by
+    DPU index. Platforms that require to overwrite the platform.json file
+    will use this API
+
+    This is valid only on the Switch and not on DPUs
+
+    Args:
+        index: An integer, the index of the module to retrieve
+
+    Returns:
+        DPU Port: A string Ex: For index:0 "dpu0", index:1 "dpu1"
+        See the NPU to DPU port mapping
+```
+#### 3.1.3 NPU to DPU port mapping
+platform.json of NPU/switch will show the NPU port to DPU port mapping. This will be used by services early in the system boot for midplane IP assignment. 
+```
+"DPUs" : [
+    {
+      "dpu0": {
+                "midplane_interface":  "dpu0"
+       }
+    },
+    {
+       "dpu1": {
+                "midplane_interface":  "dpu1"
+        },
+    },
+    .
+    .
+    {
+       "dpuX": {
+                "midplane_interface":  "dpux"
+        },
+    },
+
+]
+```
+On the DPU's platform.json, we can have 
+```
+DPU: {
+
+  // Anything specific to DPU, else remain empty
+
+}
+```
+#### 3.1.4 ModuleBase class API enhancements
+get_base_mac(self):
+```
+    Retrieves the base MAC address for the module
+
+    Returns:
+        A string containing the MAC address in the format 'XX:XX:XX:XX:XX:XX'
+```
+
+get_system_eeprom_info(self):
+```
+    Retrieves the full content of system EEPROM information for the DPU module
+
+    Returns:
+        DPU modules return a dictionary with key value pairs of IDPROM 
+```
+
+get_name(self):
+```
+    Retrieves the name of the device
+
+    Returns:
+        string: The name of the device. Ex; SWITCH, DPU0, DPUX
+```
+
+get_description(self):
+```
+    Retrieves the platform vendor's product description of the module
+
+    Returns:
+        A string, providing the vendor's product description of the module.
+```
+
+get_slot(self):
+```
+    Retrieves the DPU ID in case of smartswitch
+
+    Returns:
+        An integer, indicating the slot number Ex: 0:SWITCH, 1:DPU1, X:DPUX
+```
+
+get_type(self):
+```
+    Retrieves the type of the module.
+
+    Returns:
+        A string, the module-type from one of the predefined types:
+        MODULE_TYPE_SWITCH, MODULE_TYPE_DPU
+```
+
+get_oper_status(self):
+```
+    Retrieves the operational status of the module
+    This information is not sufficient for debugging complex DPU failures.
+    So, couple of new CLIs will be introduced.
+
+    Returns:
+        A string, the operational status of the module from one of the
+        predefined status values: MODULE_STATUS_EMPTY, MODULE_STATUS_OFFLINE,
+        MODULE_STATUS_FAULT, MODULE_STATUS_PRESENT or MODULE_STATUS_ONLINE
+```
+
+reboot(self, reboot_type):
```

Comment:
@rameshraghupathy i am not sure of the use case. This looks to be very platform specific.

Reply:
This is for booting each DPU module and it is platform common.

Reply:
Agree with @prgeor. The `reboot_type` is very platform-specific. I suggest to remove this parameter